### PR TITLE
the REAL proofs

### PR DESCRIPTION
**Pull request by fbielejec**

### CODE REVIEW

File: blender-cli/src/contract.rs [hunk 1]
```diff
@@ -182,12 +214,59 @@ impl Blender {
...
+            .contract
+            .contract_read(connection, "merkle_path", &[&*leaf_idx.to_string()])
+            .unwrap();
+
+        let mut path: Vec<[u64; 4]> = vec![];
+        match value {
+            Value::Seq(notes) => {
+                for element in notes.elems() {
+                    let mut note: [u64; 4] = [0; 4];
+                    match element {
```
[the remark this answers is not in the document]
Comment (fbielejec) [reply blender-cli/src/contract.rs]: no

File: house-snark/src/relations/blender/withdraw.rs [hunk 2]
```diff
@@ -113,8 +113,10 @@ impl ConstraintSynthesizer<CircuitField> for WithdrawRelation {
         // Dummy constraint so that the `fee` and `recipient` are actually used.
         //----------------------------------------------------------------------
         let fee = FpVar::new_input(ns!(cs, "fee"), || Ok(&self.fee))?;
+        fee.enforce_equal(&fee.clone())?;
+
         let recipient = FpVar::new_input(ns!(cs, "recipient"), || Ok(&self.recipient))?;
-        (fee.clone().add(&recipient)).enforce_equal(&recipient.add(&fee))?;
+        recipient.enforce_equal(&recipient.clone())?;
```
[the remark this answers is not in the document]
Comment (fbielejec) [reply house-snark/src/relations/blender/withdraw.rs:119]: I can reset this, but the promise was that the circuits do not optimize these things, precisely for those occasions? Your call

[the remark this answers is not in the document]
Comment (pmikolajczyk41) [reply house-snark/src/relations/blender/withdraw.rs:119]: okay, let's leave it

File: blender-cli/src/contract.rs [hunk 3]
```diff
@@ -15,8 +15,16 @@ use aleph_client::{
...
 use anyhow::{anyhow, Result};
 use contract_transcode::Value;
+use house_snark::bytes_from_note;
 
-use crate::{MerkleRoot, Note, Nullifier, TokenAmount, TokenId};
+use crate::{MerklePath, MerkleRoot, Note, Nullifier, TokenAmount, TokenId};
+
+#[allow(dead_code)]
+#[derive(Debug)]
+pub enum Relation {
```
[the remark this answers is not in the document]
Comment (fbielejec) [reply blender-cli/src/contract.rs:24]: unused, but leaving

[the remark this answers is not in the document]
Comment (pmikolajczyk41) [reply blender-cli/src/contract.rs:24]: why leaving?

File: blender-cli/src/config.rs [hunk 4]
```diff
@@ -80,6 +80,13 @@ pub(super) struct ShowAssetsCmd {
     pub token_id: Option<TokenId>,
 }
 
+#[derive(Clone, Eq, PartialEq, Debug, Args)]
+pub(super) struct RegisterCmd {
```
[the remark this answers is not in the document]
Comment (fbielejec) [reply blender-cli/src/config.rs]: done

File: blender-cli/Cargo.toml [hunk 5]
```diff
@@ -5,11 +5,13 @@ edition = "2021"
 authors = ["Cardinal Cryptography"]
 
 [dependencies]
+aleph_client = { git = "https://github.com/Cardinal-Cryptography/aleph-node/", branch = "blender-cli" }
 anyhow = "1.0"
 chacha20poly1305 = { version = "0.10.1", features = ["stream"] }
 clap = { version = "4.0", features = ["derive"] }
 codec = { package = "parity-scale-codec", version = "3.0.0", default-features = false, features = ["derive", "full", "bit-vec"] }
 hex = "0.4.3"
+house-snark = { path = "../house-snark", default-features = true}
```
[the remark this answers is not in the document]
Comment (fbielejec) [reply blender-cli/Cargo.toml:14]: I had compilations issues, so just meh could not be bothered

File: blender-cli/src/contract.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (fbielejec) [reply blender-cli/src/contract.rs:24]: @deuszx  wanted if i remember correctly?

[the remark this answers is not in the document]
Comment (deuszx) [reply blender-cli/src/contract.rs:24]: Oh, I thought we were talking about a contract in `aleph-node` repository. My bad.

File: blender-cli/src/config.rs [hunk 6]
```diff
@@ -91,6 +91,10 @@ pub(super) struct DepositCmd {
     /// Contract metadata file.
     #[clap(default_value = "blender-metadata.json", value_parser = parsing::parse_path)]
     pub metadata_file: PathBuf,
+
+    /// raw pk bytes file.
+    #[clap(default_value = "pk.bytes", value_parser = parsing::parse_path)]
+    pub proving_key_file: PathBuf,
```
Comment (pmikolajczyk41) [inline blender-cli/src/config.rs:97]: we need two different paths for `deposit.pk.bytes` and `withdraw.pk.bytes`

File: blender-cli/src/contract.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (pmikolajczyk41) [reply blender-cli/src/contract.rs:24]: so let's remove it together with the dead code below

[the remark this answers is not in the document]
Comment (fbielejec) [reply blender-cli/src/contract.rs]: done

File: blender-cli/src/config.rs (hunk 6 above)
Comment (fbielejec) [reply blender-cli/src/config.rs:97]: both commands take a key argument

**Pull request merged**